pkg/kubelet/kuberuntime/kuberuntime_image.go

```diff
@@ -32,11 +32,8 @@ import (
 // secrets if necessary.
 func (m *kubeGenericRuntimeManager) PullImage(ctx context.Context, image kubecontainer.ImageSpec, pullSecrets []v1.Secret, podSandboxConfig *runtimeapi.PodSandboxConfig) (string, error) {
 	img := image.Image
-	repoToPull, _, _, err := parsers.ParseImageName(img)
-	if err != nil {
-		return "", err
-	}
-
+	// Ignore error since we are validating the image names when creating these pods.
```

Comment:
this is strange. If `err` is not `nil`, it will indicate that the `repoToPull` may not contain valid information. This may lead to all sorts of trouble

Reply:
yea I think I was a bit eager yesterday.  I think we can't remove the validation but I tried to at least standarize on one approach for `ParseImageName`.